app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -314,12 +314,36 @@ class FormController(
         form = cache.form.copy(visitsIndex = VisitIndex(visitsIndex), componentIdToFileId = componentIdToFileId)
       )
 
-      validateAndUpdateData(cacheUpd, processDataUpd, sn, sn, maybeAccessCode, ff, formModelOptics) { _ =>
-        val sectionTitle4Ga = getSectionTitle4Ga(processDataUpd, sn)
-        Redirect(
-          routes.FormController
-            .form(cache.formTemplate._id, maybeAccessCode, sn, sectionTitle4Ga, SuppressErrors.Yes, FastForward.Yes)
-        )
+      validateAndUpdateData(cacheUpd, processDataUpd, sn, sn, maybeAccessCode, ff, formModelOptics) {
+        maybeSectionNumber =>
+          val sectionTitle4Ga = getSectionTitle4Ga(processDataUpd, sn)
+
+          val sectionNumber =
+            if (
+              formModelOptics.formModelRenderPageOptics.formModel.brackets
+                .addToListBracket(addToListId)
+                .repeaters
```

Comment:
Use `.iterations` instead of `.repeaters` (Look at `repeaters` to see why it is better)